File: Projects/DrugTraceability/QR-Codes/simulation/sim4/knowledge/intermediate/inter1/database/queries.py
from .connection import get_connection

def get_pack_by_id(pack_id):
    conn = get_connection()
    c = conn.cursor()
    c.execute('SELECT name FROM packs WHERE id = ?', (pack_id,))
    pack = c.fetchone()
    conn.close()
    return pack

def get_box_by_id(box_id):
    conn = get_connection()
    c = conn.cursor()
    c.execute('SELECT name, pack_id FROM boxes WHERE id = ?', (box_id,))
    box = c.fetchone()
    conn.close()
    return box

def get_product_by_id(product_id):
    conn = get_connection()
    c = conn.cursor()
    c.execute('SELECT name, box_id FROM products WHERE id = ?', (product_id,))
    product = c.fetchone()
    conn.close()
    return product

File: Projects/DrugTraceability/QR-Codes/simulation/sim4/knowledge/advanced/adv3/core/queries.js
// Query strings for hierarchical module

// Create Hierarchy
// get product name of the manufacturer
const getProductByManufacturerName = "SELECT product_name FROM companies WHERE username = $1";
const getManufacturereId = "SELECT id FROM users WHERE username = $1";
const getRole = "SELECT role FROM users WHERE username = $1";
const getCompanyId = "SELECT id FROM companies WHERE username = $1";

// insert pack
const insertPack = "INSERT INTO packs (name, company_id) VALUES ($1, $2) RETURNING id";

// insert box 
const insertBox = "INSERT INTO boxes (name, pack_id, company_id) VALUES ($1, $2, $3) RETURNING id";

// insert product
const insertProduct = "INSERT INTO products (product_name, identity_number, pack_id, box_id, company_id) VALUES ($1, $2, $3, $4, $5) RETURNING id";


// Fetch Hierarchy
// get all packs
const getAllPacks = "SELECT id, name FROM packs";

// get box name by pack ID
const getBoxNameByPackId = "SELECT id, name FROM boxes WHERE pack_id = $1";

// get product number by box ID
const getProductNameByBoxId = "SELECT id, product_name FROM products WHERE box_id = $1";


// Print Hierarchy
// get company info 
const getCompanyInfo = "SELECT name, address FROM companies WHERE id = $1" ;

// get packs by company ID and date
const getPacksByCompanyAndDate = "SELECT id FROM packs WHERE company_id = $1 AND DATE(created_at) = $2";

// get boxes by pack ID
const getBoxesByPackId = "SELECT id FROM boxes WHERE pack_id = $1";

// get products by box ID
const getProductsByBoxId = "SELECT id FROM products WHERE box_id = $1";


module.exports = {
    getProductByManufacturerName,
    getManufacturereId,
    getRole,
    getCompanyId,
    insertPack,
    insertBox,
    insertProduct,
    getAllPacks,
    getBoxNameByPackId,
    getProductNameByBoxId,
    getCompanyInfo,
    getPacksByCompanyAndDate,
    getBoxesByPackId,
    getProductsByBoxId
}